From f73cbf1333c0788b98ff4e47cf6270781a7fa6b1 Mon Sep 17 00:00:00 2001
From: Eric Lundby <Eric.Lundby@gmail.com>
Date: Mon, 11 May 2026 15:31:00 -0600
Subject: [PATCH] Mark KDTree stub as disjoint base

Newer mypy stubtest versions detect Cython extension classes that are
disjoint bases at runtime. Mark KDTree accordingly so the installed stub
matches the runtime class behavior.
---
 pykdtree/kdtree.pyi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pykdtree/kdtree.pyi b/pykdtree/kdtree.pyi
index 9dc6fe844b374ad5318bade732f63365651dc1c1..8aa302b9d95045929fc9b97e00c05aeb1eb62147 100644
--- a/pykdtree/kdtree.pyi
+++ b/pykdtree/kdtree.pyi
@@ -14,8 +14,10 @@
 # You should have received a copy of the GNU Lesser General Public License along
 # with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+from _typeshed import disjoint_base
 from typing import Any
 import numpy as np
 
+@disjoint_base
 class KDTree:
     """kd-tree for fast nearest-neighbour lookup.
 
-- 
2.50.1 (Apple Git-155)
